lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -47,6 +50,14 @@ defmodule Mix.Tasks.Compile.Elixir do
   def run(args) do
     {opts, _, _} = OptionParser.parse(args, switches: @switches)
 
+    # The :long_compilation_threshold is in seconds, but we want to pass it to
+    # the compiler in milliseconds.
+    opts =
+      case Keyword.fetch(opts, :long_compilation_threshold) do
```

Comment:
This would require us to change https://github.com/elixir-lang/elixir/blob/ed8aac2fbbca9327b2492eacc167bcb3bdc34690/lib/elixir/lib/kernel/parallel_compiler.ex#L117 to become:

```elixir
timeout = Keyword.get(options, :long_compilation_threshold) || 5_000
```

I think I like the more explicit one better, but yeah, we can make those changes.